File: Data Structures & Algorithms/maximum-subarray/submission-1.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int current_sum=nums[0];
        int ans=nums[0];
        for(int i=1;i<nums.size();i++){
            if(current_sum<0){
                current_sum=0;
            }
            current_sum+=nums[i];
            ans=max(ans,current_sum);
        }
        return ans;
    }
};

File: Data Structures & Algorithms/trapping-rain-water/submission-1.js
class Solution {
    /**
     * @param {number[]} height
     * @return {number}
     */
    trap(height) {
        let leftMaxArr=[0];
        let rightMaxArr=[];
        let leftMax=0;
        for(let i=0;i<height.length;i++){
            leftMaxArr[i]=leftMax;
            leftMax= Math.max(leftMax,height[i]);
        }
        let rightMax=0;
        for(let i=height.length-1;i>=0;i--){
            rightMaxArr[i]=rightMax;
            rightMax= Math.max(rightMax,height[i]);
        }
        let ans=0;
    
        for(let i=0;i<height.length;i++){
            if(height[i]<leftMaxArr[i] &&
            height[i]<rightMaxArr[i]){
                ans+=Math.min(leftMaxArr[i],rightMaxArr[i])-height[i];
            }
        }
        return ans;
    }
}
